sigma/backends/elasticsearch/elasticsearch_eql.py

```diff
@@ -0,0 +1,445 @@
+import re
+import json
+from typing import ClassVar, Dict, List, Optional, Pattern, Tuple, Union, Any
+
+from sigma.conversion.state import ConversionState
+from sigma.rule import SigmaRule
+from sigma.conversion.base import TextQueryBackend
+from sigma.conversion.deferred import DeferredQueryExpression
+from sigma.conditions import (
+    ConditionItem,
+    ConditionAND,
+    ConditionOR,
+    ConditionNOT,
+    ConditionFieldEqualsValueExpression,
+)
+from sigma.types import SigmaCompareExpression, SigmaNull, SpecialChars, SigmaNumber
+import ipaddress
+import sigma
+
+
+class EqlBackend(TextQueryBackend):
+    """
+    Elasticsearch event querly language backend. Generates query strings described here in the
+    Elasticsearch documentation:
+
+    https://www.elastic.co/guide/en/elasticsearch/reference/current/eql.html
+    """
+
+    # A descriptive name of the backend
+    name: ClassVar[str] = "Elasticsearch EQL"
+    # Output formats provided by the backend as name -> description mapping.
+    # The name should match to finalize_output_<name>.
+    formats: ClassVar[Dict[str, str]] = {
+        "default": "Plain Elasticsearch EQL queries.",
+        "eqlapi": "Plain EQL queries ready for '/_eql/search' API endpoint.",
+        "siem_rule": "Elasticsearch EQL queries as SIEM Rule.",
+        "siem_rule_ndjson": "Elasticsearch EQL Query as SIEM Rules in NDJSON Format.",
+    }
+    # Does the backend requires that a processing pipeline is provided?
+    requires_pipeline: ClassVar[bool] = True
+
+    # Operator precedence: tuple of Condition{AND,OR,NOT} in order of precedence.
+    # The backend generates grouping if required
+    precedence: ClassVar[Tuple[ConditionItem, ConditionItem, ConditionItem]] = (
+        ConditionNOT,
+        ConditionOR,
+        ConditionAND,
+    )
+    # Expression for precedence override grouping as format string with {expr} placeholder
+    group_expression: ClassVar[str] = "({expr})"
+    parenthesize: bool = True
+
+    # Generated query tokens
+    token_separator: str = " "  # separator inserted between all boolean operators
+    or_token: ClassVar[str] = "or"
+    and_token: ClassVar[str] = "and"
+    not_token: ClassVar[str] = "not"
+    # Token inserted between field and value (without separator)
+    eq_token: ClassVar[str] = ":"
```

Comment:
pySigma 0.10 added support for case-sensitive matching that could be integrated here with the `==` operator. See [base backend code](https://github.com/SigmaHQ/pySigma/blob/485773b0557d334a006a45a5b046fbe22bd9a596/sigma/conversion/base.py#L628) for more details on this.